### PR TITLE
brlcad: init at 7.34.0

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

Add package [BRL-CAD](https://brlcad.org/).
I set the license to `licenses.free` because they seem to use a combination of LGPL, BSD and public domain licenses.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>brlcad</li>
  </ul>
</details>

File: pkgs/applications/graphics/brlcad/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  cmakeFlags = [
+    "-DBRLCAD_ENABLE_STRICT=OFF"
+    "-DCMAKE_BUILD_TYPE=Release"
+  ];
+
+  meta = with lib; {
+    homepage = "https://brlcad.org";
+    description = "BRL-CAD is a powerful cross-platform open source combinatorial solid modeling system";
+    license = licenses.free;
```
Comment (wegank) [inline pkgs/applications/graphics/brlcad/default.nix:43]:
```suggestion
    license = licenses.lgpl21Plus;
```

Or even a list?

Comment (GaetanLepage) [reply pkgs/applications/graphics/brlcad/default.nix]: done

File: pkgs/applications/graphics/brlcad/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "brlcad";
+  version = "7.34.0";
+
+  src = fetchFromGitHub {
+    owner = "BRL-CAD";
+    repo = pname;
+    rev = "refs/tags/rel-7-34-0";
```
Comment (wegank) [inline pkgs/applications/graphics/brlcad/default.nix:19]:
```suggestion
    rev = "refs/tags/rel-${lib.replaceStrings [ "." ] [ "-" ] version}";
```

Comment (GaetanLepage) [reply pkgs/applications/graphics/brlcad/default.nix]: Done !

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 224810` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>brlcad</li>
  </ul>
</details>

**Pull request merged**